src/clients/SpokePoolClient/EVMSpokePoolClient.ts

```diff
@@ -0,0 +1,234 @@
+import { Contract, EventFilter } from "ethers";
+import {
+  BigNumber,
+  DepositSearchResult,
+  getDepositIdAtBlock,
+  getNetworkName,
+  InvalidFill,
+  MakeOptional,
+  toBN,
+} from "../../utils";
+import {
+  EventSearchConfig,
+  paginatedEventQuery,
+  sortEventsAscendingInPlace,
+  spreadEventWithBlockNumber,
+} from "../../utils/EventUtils";
+import { isUpdateFailureReason } from "../BaseAbstractClient";
+import { knownEventNames, SpokePoolClient, SpokePoolUpdate } from "./SpokePoolClient";
+import winston from "winston";
+import { HubPoolClient } from "../HubPoolClient";
+import { SpokePool } from "../../typechain";
+import {
+  findDepositBlock,
+  getMaxFillDeadlineInRange as getMaxFillDeadline,
+  getTimeAt as _getTimeAt,
+  relayFillStatus,
+  isZeroAddress,
+} from "../../utils/SpokeUtils";
+import { DepositWithBlock, FillStatus, RelayData } from "../../interfaces";
+
+/**
+ * An EVM-specific SpokePoolClient.
+ */
+export class EVMSpokePoolClient extends SpokePoolClient {
+  constructor(
+    logger: winston.Logger,
+    public readonly spokePool: Contract,
+    hubPoolClient: HubPoolClient | null,
+    chainId: number,
+    deploymentBlock: number,
+    eventSearchConfig: MakeOptional<EventSearchConfig, "toBlock"> = { fromBlock: 0, maxBlockLookBack: 0 }
+  ) {
+    super(logger, hubPoolClient, chainId, deploymentBlock, eventSearchConfig);
+  }
+
+  public async getTimestampForBlock(blockTag: number): Promise<number> {
+    const block = await this.spokePool.provider.getBlock(blockTag);
+    return Number(block.timestamp);
+  }
+
```

Comment:
This is safe to delete - it's not in master anymore ([commit](https://github.com/across-protocol/sdk/commit/43965c7ca453787ab528633643c22fcd563ac853)).

```suggestion
```

Reply:
https://github.com/across-protocol/sdk/pull/935